src/gui/dialogs/migrate_version_selection.cpp

```diff
@@ -97,19 +99,55 @@ void migrate_version_selection::post_show(window& window)
 		// given self-compilation and linux distros being able to do whatever they want plus command line options to
 		// alter locations make sure the directories/files are actually different before doing anything with them
 		if(migrate_addons_dir != filesystem::get_addons_dir()) {
+			std::vector<std::string> old_addons;
+			std::vector<std::string> current_addons;
 			std::vector<std::string> migrate_addons;
-			filesystem::get_files_in_dir(migrate_addons_dir, nullptr, &migrate_addons);
+			filesystem::get_files_in_dir(migrate_addons_dir, nullptr, &old_addons);
+
+			// if not the first time this is being run, then make sure to remove any already existing add-ons from the list
+			if(!first_time_) {
```

Comment:
Why not do this regardless? If there are addons in the dir then it seems better to always consider them.

Reply:
The first time this is run it's guaranteed not to find any though.

Reply:
How would the player had add-ons installed before they first start wesnoth to create the add-ons folder?

Reply:
You can install and uninstall wesnoth. You can create/delete/copy directories...

Reply:
Whatever, updated. Reinstalling wesnoth would not cause the migrator to be rerun though since it wouldn't remove userdata-related folders. The idea that someone would manually create the wesnoth userdata folder structure before installing, while technically possible to happen, is still fairly absurd.